src/main/java/com/thealgorithms/sorts/SelectionSort.java

```diff
@@ -1,46 +1,38 @@
 package com.thealgorithms.sorts;
 
+/**
+ * This class provides an implementation of the selection sort algorithm.
+ * It sorts an array of elements in increasing order using an iterative approach.
+ */
 public class SelectionSort implements SortAlgorithm {
-
     /**
-     * Generic selection sort algorithm in increasing order.
+     * Sorts an array of comparable elements in increasing order using the selection sort algorithm.
      *
-     * @param arr the array to be sorted.
-     * @param <T> the class of array.
-     * @return sorted array.
+     * @param array the array to be sorted
+     * @param <T> the class of array elements
+     * @return the sorted array
      */
     @Override
-    public <T extends Comparable<T>> T[] sort(T[] arr) {
-        int n = arr.length;
-        for (int i = 0; i < n - 1; i++) {
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        if (array.length == 0) {
+            return array;
+        }
+
+        // One by one move the boundary of the unsorted subarray
+        for (int i = 0; i < array.length - 1; i++) {
+            // Find the minimum element in the unsorted array
             int minIndex = i;
-            for (int j = i + 1; j < n; j++) {
-                if (arr[minIndex].compareTo(arr[j]) > 0) {
+            for (int j = i + 1; j < array.length; j++) {
+                if (array[j].compareTo(array[minIndex]) < 0) {
                     minIndex = j;
                 }
             }
```

Comment:
This deserves its own method.

Reply:
This also already made in this suggestion https://github.com/TheAlgorithms/Java/pull/5265#discussion_r1659318388